crates/amaru/src/sync/pull.rs

```diff
@@ -86,6 +88,45 @@ impl Stage {
         let _intersection = point.ok_or(miette!("couldn't find intersect")).or_panic()?;
         Ok(())
     }
+
+    pub async fn roll_forward(&mut self, header: &HeaderContent) -> Result<(), WorkerError> {
+        let span_forward = debug_span!(
+            target: EVENT_TARGET,
+            "pull.roll_forward",
+            header.slot = tracing::field::Empty,
+            header.hash = tracing::field::Empty,
+            peer = self.peer_session.peer.name
+        );
```

Comment:
Don't you need to _enter_ the span here 🤔 ?

Reply:
Actually no :) The docs from tracing actually say `enter` should be avoided for async code, and there's some automatic exit happening when the span goes out of scope

Reply:
🤯 ... too much docs to read